### PR TITLE
Mattermost Alert Transport

### LINKED ISSUE #5659: Feature Request: Notifications To Mattermost
The option to have alerts/notifications sent to Mattermost would be nice.

### PR DESCRIPTION
**Pull request by gpant**
Added support for Mattermost notifications (Mattermost Transport).

George Pantazis <gpant@eservices-greece.com>

fixes #5659

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
I thought most people used the Slack transport for this?

**Comment by gpant** [comment]
You could use the slack transport but you would miss the mattermost specific features.

File: LibreNMS/Alert/Transport/Mattermost.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+use LibreNMS\Alert\Transport;
+
+class Mattermost extends Transport
+{
+    public function deliverAlert($obj, $opts)
+    {
+        if (empty($this->config)) {
+            return $this->deliverAlertOld($obj, $opts);
+        }
+        $mattermost_opts['url'] = $this->config['mattermost-url'];
```
Comment (murrant) [inline LibreNMS/Alert/Transport/Mattermost.php]: This array should be initialized.

Either as:
```php
$mattermost_opts = [
     'url' => $this->config['mattermost-url'],
     ...
];
```

or

```php
$mattermost_opts = [];
$mattermost_opts['url'] = $this->config['mattermost-url'];
...
```

File: LibreNMS/Alert/Transport/Mattermost.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($curl, CURLOPT_POST, true);
+        curl_setopt($curl, CURLOPT_POSTFIELDS, $alert_payload);
+
+        $ret = curl_exec($curl);
+        $code = curl_getinfo($curl, CURLINFO_HTTP_CODE);
+        if ($code != 200) {
+            d_echo("Mattermost Connection Error: " . $ret);
+            return false;
+            return 'HTTP Status code ' . $code;
```
Comment (murrant) [inline LibreNMS/Alert/Transport/Mattermost.php]: remove the second return line here, it is unreachable.

**Comment by Tatermen** [comment]
This commit deleted .htaccess from the html folder, rendering the web UI completely inoperable in Apache.

**Comment by empi89** [comment]
@Tatermen Thanks for pointing this out. I have just seen this as well. There were files modified that are completely unrelated to the subject of this PR. 

Would be great to review the pull request approval process of this project. In combination with the daily updates by default things like this have a negative impact on the project as a whole.

**Pull request merged**